src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -968,7 +968,7 @@ private bool HasCovariantArrayConversion(TypeSymbol source, TypeSymbol destinati
             // * Both SE and TE are reference types.
             // * An implicit reference conversion exists from SE to TE.
             return
-                (s.Rank == d.Rank) &&
+                (s.Rank == d.Rank && s.IsSZArray == d.IsSZArray) &&
```

Comment:
Couldn't this just be simplified to `s.IsSZArray == d.IsSZArray`?  I don't understand how this could be true and the ranks could be different.